frontend/src/models/Deposit.ts

```diff
@@ -0,0 +1,15 @@
+export interface IDepositController {
+    getDeposits: () => Promise<Deposit[]>
+    getDeposit: (depositId: string) => Promise<Deposit>
+    getLatestDeposits: () => Promise<Deposit[]>
+    addDeposit: (deposit: Deposit) => Promise<Deposit[]>
+    changeDeposit: (deposit: Deposit) => Promise<Deposit[]>
+    deleteDeposit: (depositId: string) => Promise<Deposit[]>
+}
+
+export interface Deposit {
+    depositId?: string
```

Comment:
Why is your id optional? If its only because of your creation process, i would create for the creation process a seperate model like DepositCreationDto